orc8r/cloud/go/mproto/mproto_test.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mproto_test
+
+import (
+	b64 "encoding/base64"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"magma/orc8r/cloud/go/mproto"
+	mocks "magma/orc8r/cloud/go/mproto/mocks"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	iterationCount = 1000
+)
+
+var (
+	testdataDir    = "testdata"
+	goldenFilepath = filepath.Join(testdataDir, "determinisic-digest.b64.golden")
+)
+
+// TestMarshalManyDeterministic checks if MarshalManyDeterministic truly enforces
+// deterministic encoding by comparing encoded protobuf messages containing the same data
+func TestMarshalManyDeterministic(t *testing.T) {
+	// Encode basic proto messages (no compound fields)
+	testDataObjs := getTestDataBasic()
+	protos1 := map[string]proto.Message{
+		testDataObjs[0].Key: &testDataObjs[0],
+		testDataObjs[1].Key: &testDataObjs[1],
+		testDataObjs[2].Key: &testDataObjs[2],
+	}
+	protos2 := map[string]proto.Message{
+		testDataObjs[0].Key: &testDataObjs[0],
+		testDataObjs[1].Key: &testDataObjs[1],
+		testDataObjs[2].Key: &testDataObjs[2],
+	}
+
+	for i := 0; i < iterationCount; i++ {
+		encoded1, err1 := mproto.MarshalManyDeterministic(protos1)
+		assert.NoError(t, err1)
+		encoded2, err2 := mproto.MarshalManyDeterministic(protos2)
+		assert.NoError(t, err2)
+		assert.Equal(t, encoded1, encoded2)
+	}
+
+	// Encode compound proto messages (with repeated & map fields)
+	testDataCollections := getTestDataCompound()
+	protos3 := map[string]proto.Message{
+		testDataCollections[0].Id: &testDataCollections[0],
+		testDataCollections[1].Id: &testDataCollections[1],
+		testDataCollections[2].Id: &testDataCollections[2],
+	}
+	protos4 := map[string]proto.Message{
+		testDataCollections[0].Id: &testDataCollections[0],
+		testDataCollections[1].Id: &testDataCollections[1],
+		testDataCollections[2].Id: &testDataCollections[2],
+	}
+
+	for i := 0; i < iterationCount; i++ {
+		encoded3, err3 := mproto.MarshalManyDeterministic(protos3)
+		assert.NoError(t, err3)
+		encoded4, err4 := mproto.MarshalManyDeterministic(protos4)
+		assert.NoError(t, err4)
+		assert.Equal(t, encoded3, encoded4)
+	}
```

Comment:
This is slightly wonky. Just do it once, then 1000 times do it once each and compare to that original encoded value. Sort of a do-while loop